### DISCUSSION TITLE
😎 Custom emoji support per orginization.

### DISCUSSION THREAD

**Discussion by BlueFalconHD** [community/community#32258] [Other Feature Feedback, Questions, & Ideas]
## ❌ Difference from #16150
This issue goes more in depth about the implementation, the benefits, the problems, alternative solutions, and more. I have read over this issue but feel that this is not a duplicate, but an innovation.

## ⚙️ Implementation
This feature could be implemented in one of two ways.
1. There could be a section in organization settings dedicated to custom emojis. This would be kind of like Discord's custom emoji implementation. A button to upload emoji, and a list of all current custom emojis. Each list item would have a way to change the emoji's alias, so you could change it from `:obscure-image-name156745:` to `:kekw:`. There would also be a button to simply remove the emoji.
2. An implementation revolving around the org's `.github` repository. This would use a `.yml` file, maybe `emojis.yml`, along with a folder for the actual emoji images. The real benefit with this option is the ability for people to open a PR or an issue requesting new emoji.

## ✅ Benefits
The benefit of this would be the ability to have a tighter knit community and a way to get universal symbols for things like the contribution emoji key.

## 🫠 Problems
The implementation of this has 2 problems that are clear:
1. Scope
> Should the emojis for any org you are in be active anywhere, or just in the org?
2. Limit
> GitHub might need to add a limit depending on how custom emoji are implemented. If a community has hundreds of emoji, it might crowd the emoji picker.

## 📖 Other
Slack has implemented this, and it is a loved feature. This should not directly affect GitHub's decision, but the community should. If a community has some emoji on Discord and or Slack, it would be great to parity that to GitHub. For example, on the [catppuccin](https://github.com/catppuccin) Discord server, we have this emoji:
<img width="41" alt="image" src="https://user-images.githubusercontent.com/72631767/189002505-9df16850-e51b-490e-9101-baddcbaf3eaf.png">
It is really useful for showing interest in a PR or discussion, and it would be really nice to get it on our GitHub, as it already is on our Discord and is widely used.

Also, another thing related would be a way of making a custom emoji a reaction to releases, discussions, issues, PRs, and any other reactionable things.

## 🔚 Conclusion
In conclusion, I believe that custom emoji would benefit GitHub in many ways.

## ⏰ Changelog
> **Wednesday 7, September at 7:04pm CDT**
Edited the end of [📖 Other](##-📖-Other) to include some sentences I forgot to type :shipit:

Thanks!
\- blue

**Comment by jxu** [comment]
Why do you keep referencing Discord and Slack? GitHub does not serve that purpose.

**Reply by BlueFalconHD** [reply]
It is a great feature. While Discord or Slack do not completely align with GitHub, it would be a cool feature to come to GitHub, and those are some platforms that already have this feature.
